src/main/java/magus/console/Parser.java

```diff
@@ -0,0 +1,253 @@
+package magus.console;
+
+import magus.exception.ArgumentNotFoundException;
+import magus.exception.UnknownArgumentException;
+import magus.util.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Parses user input from console into command and additional input
+ */
+public class Parser {
+    private final String commandCandidate;
+    private final Command command;
+    private final String additionalInput;
+    private final List<String> additionalInputList;
+
+    /**
+     * Constructor for Parser that parses the user input as such: first word is the Command
+     * and remaining words are additional input
+     *
+     * @param input User input from console
+     */
+    public Parser(String input) {
+        // Parse first word to get command
+        String firstWord = input;
```

Comment:
misleading comment. you are assigning the entire input to `firstWord`.
additionally, `firstWord` is being used to manipulate the input. better to have a temporary variable to make it explicit.

Reply:
Resolved in 8269dc2